sanic/request.py

```diff
@@ -356,28 +357,35 @@ def parse_multipart_form(body, boundary):
             )
 
             if form_header_field == "content-disposition":
-                file_name = form_parameters.get("filename")
                 field_name = form_parameters.get("name")
+                file_name = form_parameters.get("filename")
+
+                # non-ASCII filenames in RFC2231, "filename*" format
+                if file_name is None and form_parameters.get("filename*"):
```

Comment:
Since the `form_parameters` lookup is being done twice, maybe we can move the lookup to a single line and use a variable instead? (this is just a good to have change)

```python
filename_rfc5987 = form_parameters.get("filename*")

if not file_name and filename_rfc5987:
    encoding, _, value = email.utils.decode_rfc5987(filename_rfc2231)
    file_name = unquote(value, encoding=encoding)
```
Also the format you are using is `RFC 5987` standard and not `RFC2231` if I remember correctly. (Sorry, for the nitpicking)